evo-delivery-app/evo-delivery-backend/src/services/route.service.ts

```diff
@@ -1,12 +1,79 @@
-import {DistanceMatrix, Driver, EaEvaluateConfig, EaEvaluateResponse, Order} from "../types";
+import {Depot, DistanceMatrix, Driver, EaEvaluateConfig, EaEvaluateResponse, Order} from "../types";
 import {driverService, draftService, orderService, depotService} from '../services';
 import {googleMatrixClient, eaHttpClientAdapter} from "../clients";
+import driverModel from "../database/models/driver.model";
+import orderModel from "../database/models/order.model";
+import { Console } from "console";
 
 
 export const store = async (draftId: string, routes: EaEvaluateResponse) => {
-  return routes;
+  const draft = await draftService.getById(draftId)
+  const distances = draft.data.distances!;
+  const depotId = draft.data.depot;
+  const enricheRoutes: DriverRoute[] = []
+  const depot = await depotService.getById(draft.data.depot);
+  for (const [driverId, ordersIds] of Object.entries(routes)){
```

Comment:
More functional, and in my opinion elegant way to write this is:

  const enrichedRoutes = Object.entries(routes).map(async ([driverId, ordersIds]) => {
      const driversP = driverModel.getById(driverId);
      const ordersP = orderModel.getByIds(ordersIds)
      const [driver,orders] = await Promise.all([driversP, ordersP]);
      await enrichDriverRoute(depotId, driver, orders, distances);
    })

Reply:
done